src/Microsoft.AspNetCore.Authentication.Abstractions/AuthenticationProperties.cs

```diff
@@ -86,12 +92,12 @@ public string RedirectUri
             set => SetBool(RefreshKey, value);
         }
 
-        private string GetString(string key)
```

Comment:
Make them public? They're operating on public state.

Reply:
All of them? Don’t want it to be too noisy and distract from the properties.

Reply:
Hmm, ok, leave it protected, but it still needs a basic doc comment.

Reply:
Why are we exposing these things?

Reply:
(GitHub’s UI sometimes makes it really easy to miss comments…)

@davidfowl I made the string-based helpers public so that they can serve as a more friendly way to interact with the `Items` dictionary. And especially so that handlers have a simple API to retrieve values (for when they actually want to persist data inside `Items`).

Reply:
I thought we were going to keep these as protected?

Reply:
We don't actually need Get/SetParameter public either right?  We are going to have properties that expose the parameters right?

Reply:
We don't do this anywhere else in the stack.

Reply:
We do, but they're usually extension methods.

Reply:
Where?

Reply:
I don’t mind either way but as soon as a handler needs a value from the items collection, they are effectively going to reimplement the thing again. And since the `GetParameter` method is public, I don’t think a set of public methods for the `Items` collection hurts.

Reply:
https://github.com/aspnet/Caching/blob/dev/src/Microsoft.Extensions.Caching.Abstractions/MemoryCacheExtensions.cs